### PR TITLE
Adds font metrics to support font commands.

### PR DESCRIPTION
**Pull request by kevinb7**
This is part 1 of 3.  Rendering, screenshots, MathML, and unit tests will
follow in susbequent pull requests.

### CODE REVIEW

File: metrics/parse_tfm.py [hunk 1]
```diff
@@ -64,11 +64,14 @@ def __init__(self, start_char, end_char, char_info, width_table,
         self.ligkern_program = LigKernProgram(ligkern_table)
         self.kern_table = kern_table
 
-    def get_char_metrics(self, char_num):
+    def get_char_metrics(self, char_num, zero_indexed=False):
```
Comment (xymostech) [inline metrics/parse_tfm.py]: What does this flag do?

File: metrics/mapping.pl [hunk 2]
```diff
@@ -7,6 +7,8 @@
 # accurately extract the metrics from the corresponding .tfm (TeX font metric)
 # files
 
+# Source: https://github.com/Khan/MathJax-dev/blob/master/fonts/OTF/TeX/makeFF
```
Comment (xymostech) [inline metrics/mapping.pl]: It already says this 7 lines up!

File: metrics/extract_tfms.py [hunk 3]
```diff
@@ -31,7 +31,11 @@ def main():
         'cmsy10.tfm',
         'cmti10.tfm',
         'msam10.tfm',
-        'msbm10.tfm'
+        'msbm10.tfm',
+        'eufm10.tfm',
+        'cmtt10.tfm',
+        'rsfs10.tfm',
+        'cmss10.tfm',
```
Comment (xymostech) [inline metrics/extract_tfms.py]: :) thanks for the trailing commas

File: metrics/mapping.pl (hunk 2 above)
Comment (kevinb7) [inline metrics/mapping.pl]: I thought having the full URL was more convenient.  I'll take this out.

File: metrics/parse_tfm.py (hunk 1 above)
Comment (kevinb7) [inline metrics/parse_tfm.py]: There was one font `rsfs` where glyph indices were messed up.  This corrects them.  I'll add a docstring.

File: metrics/README.md [hunk 4]
```diff
@@ -19,3 +19,21 @@ Once you have these things, run
...
+
+Install the JSON perl module:
+    
+    sudo cpan JSON
+
+Install fontforge with python extensions (this may install python as well):
+
+    brew install fontforge --enable-pyextension
+    
+If you get a "Segmentation fault: 11" when running `make metrics` you'll have
```
Comment (xymostech) [inline metrics/README.md]: I think it maybe makes more sense to add `/usr/local/Cellar/python/2.7.9/bin/` to your path than to edit the shebang in the file?

Comment (kevinb7) [inline metrics/README.md]: It still picks up the system python.  I'd like to make this go away eventually by replacing `fontforge` with `freetype-py`.  See https://github.com/Khan/KaTeX/issues/288 for details.

Comment (gagern) [inline metrics/README.md]: The shebang has to contain a full path, the `PATH` environment variable doesn't have any effect. One common solution is writing it as

    #!/usr/bin/env python2

hoping that that's the name which is being used, and that it's on the path.

Comment (kevinb7) [inline metrics/README.md]: The system version of python is 2.7.5.  I tried `#!/usr/bin/env python2`, but it still used 2.7.5 instead of 2.7.9.

**Comment by gagern** [comment]
How shall we deal with this commit here in relation to #303? On the one hand, this commit here is the older one.

On the other hand, reviewing this here might be a lot easier if #303 got merged and this one here rebased onto that one. And there are other benefits as well. With @1f8610cebe2c99223db4082a834fb814afd2b697 you could suggest `make PYTHON=/usr/local/Cellar/python/2.7.9/bin/python` in the README, without danger of a changed shebang getting committed accidentially. You could also drop the install instructions for fontforge.

Perhaps we can simply disregard these issues about python version and fontforge installation here, merge things as they stand knowing that it's a temporary solution, then rebase #303 onto this and add a commit to that which cleans those things up. If this request here is about ready to be merged, then that solution would be preferable, since I'm really waiting for this font stuff to hit master so I can gain some traction with my subsequent restructuring plans.

**Comment by kevinb7** [comment]
I'll rebase this on to https://github.com/Khan/KaTeX/pull/303.

**Comment by kevinb7** [comment]
@xymostech this is ready for review.

**Comment by xymostech** [comment]
(Sorry for taking so long!) This all looks good now! Thanks for the docstring about rsfs, that's very strange!

**Comment by kevinb7** [comment]
Hopefully it's not just my install of TeX.

**Pull request merged**